testdata/p4_16_samples_outputs/control-hs-index-test2.p4

```diff
@@ -0,0 +1,66 @@
+#include <core.p4>
```

Comment:
These tests are a little flawed. First, the index should not be part of the ethernet stack. Use h.h.a instead. Also you should extract a in the parser. That is missing. Here is the setup I would use for all the tests you added. 

```p4
header ethernet_t {
    bit<48> dst_addr;
    bit<48> src_addr;
    bit<16> eth_type;
}

header h_stack {
    bit<8> a;
}

header h_index {
    bit<32> i;
}

struct headers {
    ethernet_t eth_hdr;
    h_stack[3] h;
    h_index    i;
}
```

with extract like this:

```p4
pkt.extract(hdr.eth_hdr);
pkt.extract(hdr.h[0]);
pkt.extract(hdr.h[1]);
pkt.extract(hdr.h[2]);
pkt.extract(hdr.i);
```